src/python_testing/TC_DeviceBasicComposition.py

```diff
@@ -560,54 +671,73 @@ class RequiredMandatoryAttribute:
+                success = False
 
-                        attribute_value = cluster[attribute_id]
-                        if isinstance(attribute_value, ValueDecodeFailure):
-                            self.record_warning(self.get_test_name(), location=location,
-                                                problem=f"Found a failure to read/decode {attribute_string} on {location.as_cluster_string(self.cluster_mapper)} when it was claimed as supported in AttributeList ({attribute_list}): {str(attribute_value)}", spec_location="AttributeList Attribute")
-                            # Warn only for now
-                            # TODO: Fail in the future
-                            continue
-                    for attribute_id in cluster:
-                        if attribute_id not in attribute_list:
-                            attribute_string = self.cluster_mapper.get_attribute_string(cluster_id, attribute_id)
-                            location = AttributePathLocation(endpoint_id, cluster_id, attribute_id)
-                            self.record_error(self.get_test_name(), location=location,
-                                              problem=f'Found attribute {attribute_string} on {location.as_cluster_string(self.cluster_mapper)} not listed in attribute list', spec_location="AttributeList Attribute")
-                            success = False
+        self.print_step(11, "Validate that standard cluster FeatureMap attributes contains only known feature flags")
+        for endpoint_id, endpoint in self.endpoints_tlv.items():
+            for cluster_id, cluster in endpoint.items():
+                if cluster_id not in chip.clusters.ClusterObjects.ALL_CLUSTERS:
+                    continue
+                feature_map = cluster[FEATURE_MAP_ID]
+                feature_mask = 0
+                try:
+                    feature_map_enum = chip.clusters.ClusterObjects.ALL_CLUSTERS[cluster_id].Bitmaps.Feature
+                    for f in feature_map_enum:
+                        feature_mask = feature_mask | f
+                except AttributeError:
+                    # If there is no feature bitmap, feature mask 0 is correct
+                    pass
+                feature_map_extras = feature_map & ~feature_mask
+                if feature_map_extras != 0:
+                    location = ClusterPathLocation(endpoint_id=endpoint_id, cluster_id=cluster_id)
+                    self.record_error(self.get_test_name(), location=location,
+                                      problem=f'Standard cluster {cluster_id} with unkonwn feature {feature_map_extras:02x}')
+                    success = False
 
         if not success:
             self.fail_current_test(
-                "At least one cluster was missing a mandatory global attribute or had differences between claimed attributes supported and actual.")
+                "At least one cluster has failed the global attribute range and support checks")
```

Comment:
I guess the above check of feature map could also result in this to fail, so maybe we need a more general failure message here?

Reply:
Ah. Yeah, wording here is weird - I was intending "global attribute" to refer to AttributeList, AcceptedCommandList, FeatureMap etc. ie, the global attributes. But yeah, it does sound like I'm only checking attributes. I'll list them out.